src/brod_consumer.erl

```diff
@@ -48,238 +42,209 @@
         , handle_info/2
         , terminate/2
         , code_change/3
-        , format_status/2
         ]).
 
 -include("brod.hrl").
 -include("brod_int.hrl").
 
--record(state, { hosts         :: [{string(), integer()}]
-               , socket        :: #socket{}
+-type consumer_option() :: offset
```

Comment:
does this imply: always have to restart the consumer process to reset offset ?
for example, if a consumer wants to search for a offset <--> timestamp (in message key) correlation, how should they implement the callback module ?